apps/test/unit/accounts/ChangeUserTypeModalTest.jsx

```diff
@@ -1,215 +1,175 @@
+      },
+    }),
+  };
+});
 
 describe('ChangeUserTypeModal', () => {
-  let wrapper;
-
-  const EMAIL_SELECTOR = 'input[type="email"]';
-  const EMAIL_OPT_IN_SELECTOR = 'select';
-
   const DEFAULT_PROPS = {
-    handleSubmit: () => {},
-    handleCancel: () => {},
+    handleSubmit: jest.fn(() => Promise.resolve()),
+    handleCancel: jest.fn(),
   };
 
-  // Helpers for selecting particular elements/components
-  const emailInput = wrapper => wrapper.find(EMAIL_SELECTOR);
-  const emailOptInSelect = wrapper => wrapper.find(EMAIL_OPT_IN_SELECTOR);
-  const submitButton = wrapper =>
-    wrapper
-      .find(Button)
-      .filterWhere(
-        n => n.prop('text') === i18n.changeUserTypeModal_save_teacher()
-      );
-  const cancelButton = wrapper =>
-    wrapper.find(Button).filterWhere(n => n.prop('text') === i18n.cancel());
+  const renderComponent = (props = {}) => {
+    return render(<ChangeUserTypeModal {...DEFAULT_PROPS} {...props} />);
+  };
 
-  beforeEach(() => {
-    wrapper = mount(<ChangeUserTypeModal {...DEFAULT_PROPS} />);
-  });
+  const getEmailInput = () => screen.getByRole('textbox', {name: /email/i});
+  const getEmailOptInSelect = () =>
+    screen.getByRole('combobox', {name: /email opt-in/i});
+  const getSubmitButton = () =>
+    screen.getByRole('button', {name: /save teacher/i});
+  const getCancelButton = () => screen.getByRole('button', {name: /cancel/i});
 
-  afterEach(() => {
-    wrapper.unmount();
+  it('renders the modal with the correct title', () => {
+    renderComponent();
+    expect(screen.getByText('Change User Type')).toBeInTheDocument();
   });
 
-  it('disables everything and shows save text when saving', () => {
-    wrapper.setState({saveState: 'saving'});
-    expect(emailInput(wrapper)).to.have.attr('disabled');
-    expect(emailOptInSelect(wrapper)).to.have.attr('disabled');
-    expect(submitButton(wrapper)).to.have.attr('disabled');
-    expect(cancelButton(wrapper)).to.have.attr('disabled');
-    expect(wrapper.text()).to.include(i18n.saving());
+  it('disables everything and shows save text when saving', async () => {
+    renderComponent();
+    fireEvent.change(getEmailInput(), {target: {value: 'valid@example.com'}});
+    fireEvent.change(getEmailOptInSelect(), {target: {value: 'yes'}});
+    fireEvent.click(getSubmitButton());
+
+    expect(getEmailInput()).toBeDisabled();
+    expect(getEmailOptInSelect()).toBeDisabled();
+    expect(getSubmitButton()).toBeDisabled();
+    expect(getCancelButton()).toBeDisabled();
+    expect(screen.getByText('Saving...')).toBeInTheDocument();
   });
 
-  it('shows unknown error text when an unknown error occurs', () => {
-    wrapper.setState({saveState: 'unknown-error'});
-    expect(wrapper.text()).to.include(
-      i18n.changeUserTypeModal_unexpectedError()
+  it('shows unknown error text when an unknown error occurs', async () => {
+    DEFAULT_PROPS.handleSubmit.mockRejectedValueOnce(
```

Comment:
I usually prefer to send in a new mocked function as I think it is clearer and doesn't change the default props (which I would consider immutable) at all. But this is fine either way.
```suggestion
    const handleSubmit = jest.fn().mockRejectedValueOnce(...)
    renderComponent({handleSubmit})
```

Reply:
I will try, I remember having some issues when mocking this specific function.